### PR TITLE
display interpreter version in console title

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/8569.

This PR allows RStudio to display the version of the currently-active interpreter in the Console pane.

Screen Shot 2021-02-09 at 2 44 05 PM

Screen Shot 2021-02-09 at 2 44 21 PM

### Approach

Relatively straight-forward listening for `reticulate` events indicating when the Python REPL has been activated or deactivated.

### Automated Tests

https://github.com/rstudio/rstudio-ide-automation/issues/185

### QA Notes

Validate that the displayed version of R + Python coincide with the "true" active interpreter. Note that merely opening a Python document is not enough to force the switch; the Python console isn't activated until Python code is actually sent to the console.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Comment by jmcphers** [comment]
It's nice to have the R version, but it feels a little weird to have the sweep button right next to the language/version indicator for R. It feels like the R version is the label for that button, or that the button will act on the R version.

Would it be too crazy to show the interpreter name instead of the word `Console`? 

```
[r icon] R 4.0.3 [separator dot] ~/foo/bar ------------------------- [broom]
```

```
[python icon] Python 3.6 [separator dot] ~/foo/bar ------------------------- 
```

**Comment by kevinushey** [comment]
That is an excellent idea!

**Comment by kevinushey** [comment]
Here's some examples. Note that one of the main challenges here is making sure the icon looks okay even on dark backgrounds, since the R + Python logos were designed to be viewed on a light background:

Screen Shot 2021-02-10 at 12 57 34 PM
Screen Shot 2021-02-10 at 12 57 08 PM
Screen Shot 2021-02-10 at 12 56 52 PM
Screen Shot 2021-02-10 at 12 56 45 PM

**Comment by kevinushey** [comment]
Here's versions with some extra pixel-smithing:

Screen Shot 2021-02-10 at 4 03 38 PM
Screen Shot 2021-02-10 at 4 03 13 PM
Screen Shot 2021-02-10 at 4 02 35 PM
Screen Shot 2021-02-10 at 4 02 23 PM
Screen Shot 2021-02-10 at 4 02 00 PM
Screen Shot 2021-02-10 at 4 01 34 PM

Feels much cleaner now!

**Comment by kevinushey** [comment]
Tweaked display for dark themes:

Screen Shot 2021-02-11 at 1 06 58 PM

Screen Shot 2021-02-11 at 1 06 26 PM

I think this PR is good to go now.

**Comment by kevinushey** [comment]
For reference, Python icon is from here and licensed as GPL-2:

https://commons.wikimedia.org/wiki/File:Python-logo-notext.svg

R logo is here:

https://www.r-project.org/logo/

and is either CC-BY-SA 4.0 or GPL-2.

File: renv.lock [hunk 1]
```diff
@@ -0,0 +1,1811 @@
+{
```
Comment (gtritchie) [inline renv.lock:1]: Is this file intended to be in the PR?

File: renv/activate.R [hunk 2]
```diff
@@ -0,0 +1,440 @@
+
```
Comment (gtritchie) [inline renv/activate.R:1]: Is this file intended to be in the PR?

**Review by gtritchie — APPROVED** [review]
Question about a couple of files (are they supposed to be in this PR) but otherwise LGTM!

File: renv.lock (hunk 1 above)
Comment (kevinushey) [reply renv.lock]: It was not! Thanks for catching both of those.

**Comment by ntluong95** [comment]
The logo does not align well, both for R and Python logo:
image
Could you please check that issues?

**Comment by kevinushey** [comment]
@ntluong95 Thanks for the heads up; I'll take a look.
